src/dc_mimefactory.rs

```diff
@@ -329,7 +324,8 @@ pub unsafe fn dc_mimefactory_load_mdn<'a>(
     );
     load_from(&mut factory);
     factory.timestamp = dc_create_smeared_timestamp(factory.context);
-    factory.rfc724_mid = dc_create_outgoing_rfc724_mid(0 as *const libc::c_char, factory.from_addr);
+    let from_addr = &factory.from_addr.as_ref().unwrap();
```

Comment:
`unwrap` is sad, but justified by the fact that `dc_create_outgoing_rfc724_mid` assumes that second argument is non-null. Previous version of code did not check for null, so I can assume that `unwrap()` is safe.